sympy/ntheory/residue_ntheory.py

```diff
@@ -452,17 +449,24 @@ def _sqrt_mod_tonelli_shanks(a, p):
     References
     ==========
 
-    .. [1] R. Crandall and C. Pomerance "Prime Numbers", 2nd Ed., page 101
+    .. [1] Carl Pomerance, Richard Crandall, Prime Numbers: A Computational Perspective,
+           2nd Edition (2005), page 101, ISBN:978-0387252827
 
     """
     s = bit_scan1(p - 1)
     t = p >> s
     # find a non-quadratic residue
-    while 1:
-        d = randint(2, p - 1)
-        r = jacobi(d, p)
-        if r == -1:
-            break
+    if p % 12 == 5:
```

Comment:
I'm not sure I understand the correlation between docstring and code. If p%8 should be 1 why are we checking whether `p%12==5` (since `5%8 != 1`)?

Reply:
I have some business to attend to this evening and probably won't be able to reply, so I just wanted to reply.

There exists a number that satisfies `p%12==5` even when `p%8==1` (e.g., 73). On the other hand, when `p%8==1`, there is no number that satisfies `p%12==7`. Therefore, the condition is `p % 12 == 5` instead of `p % 12 in [5, 7]`.

Reply:
Thanks. I see now
```python
>>> [i for i in range(1,100,8) if i%12==5]
[17, 41, 65, 89]
```

Reply:
Thanks.

And I was giving the wrong example of 73. My apologies.